woocommerce/woocommerce
# Add products data store

### All Submissions:






### Changes proposed in this Pull Request:

Adds a new products data store that makes use of the products API -> https://woocommerce.github.io/woocommerce-rest-api-docs/#list-all-products
For now this only allows for retrieving/search for products and getting the product count.

Partially addresses #32524 

### How to test the changes in this Pull Request:

1. Load this branch, create a couple products and go to **WooCommerce > Home**
2. Open your console and run `wp.data.select('wc/admin/products').getProducts({ status: 'publish'}, undefined);`
3. It should return `undefined` at first
4. Run it again, assuming the request finished, and it should return all the published products. You can try the above again with other items in the query, like `_fields: ['id']` for example, which should only return the `ids`, any query param defined here should work: https://woocommerce.github.io/woocommerce-rest-api-docs/#list-all-products
5. Now do the same for the `getProductsTotalCount` function, by running: `wp.data.select('wc/admin/products').getProductsTotalCount({ status: 'publish'}, undefined);`, this might run the correct number right away if you recently ran `getProducts` with the same query params, otherwise it should return `undefined` intiially.
6. If you have the [Redux Devtools](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en) extension, you can look for the `wc/admin/products` store and make sure the state is updated correctly and the actions are triggered correctly.
7. Run `wp.data.select('wc/admin/items').getItems('categories', { });` in your console to see if the `items` store still works correctly, and `wp.data.select('wc/admin/items').getItemsTotalCount('categories', { });`.

### Other information:




### FOR PR REVIEWER ONLY:
